### PR TITLE
docs: Add information about common problems on run command

### PR DESCRIPTION
**Pull request by ceceliacreates**
Ran into these issues when setting up my machine for Android, found these related GitHub and forum issues for solutions.
- https://github.com/ionic-team/ionic-cli/issues/3996
- https://github.com/ionic-team/native-run/issues/195
- https://forum.ionicframework.com/t/unable-to-locate-a-java-runtime-error-when-building-for-android-on-mac/221148

This pull request updates:
- Android Getting Started > Running Your App > Running on the Command-line to add note that an AVD module is needed to use the Capacitor `run android` command
- Android Troubleshooting Guide to include a solution to update `JAVA_HOME` environment variable on Mac for "Unable to locate a Java Runtime" error

### CODE REVIEW

File: docs/main/android/index.md [hunk 1]
```diff
@@ -58,6 +58,8 @@ npx cap run android
 
 The command will prompt you to select a target. [Learn more about `run`](/docs/cli/commands/run).
 
+> An Android Virtual Device (AVD) is required to use the `run` command. See the [documentation here for creating and installing AVD modules](https://developer.android.com/studio/run/managing-avds).
```
Comment (jcesarmobile) [inline docs/main/android/index.md:61]: An AVD is not required to use the run command, you can also run on a real device even if you don't have any AVD installed.
Also native-run is supposed to create one AVD if you don't have any and also there are no devices connected.

What you need is at least on emulator "system image" downloaded so native-run can create an emulator and that's usually what's missing when you run into the ERR_UNSUITABLE_API_INSTALLATION.

We have it documented on the native-run wiki

https://github.com/ionic-team/native-run/wiki/Android-Errors#err_unsuitable_api_installation

File: docs/main/android/troubleshooting.md [hunk 2]
```diff
@@ -78,6 +78,18 @@ The solution is to remove any old apps and make sure your package name is up to
 
 Finally, do a clean and rebuild just in case.
 
+## Error: "Unable to locate a Java Runtime"
+
+This error may occur when using the `run` command if the `JAVA_HOME` environment variable is not set on a Mac device.
```
Comment (jcesarmobile) [inline docs/main/android/troubleshooting.md:83]: This is not specific of Mac computers, can also happen on Windows, can you also document it for windows?

**Review by jcesarmobile — CHANGES_REQUESTED** [review]

File: docs/main/android/index.md (hunk 1 above)
Comment (ceceliacreates) [reply docs/main/android/index.md]: Ah, thanks! Updated to clarify real device or emulator system image required. The native-run wiki instructions to pass `--sdk-info` requires running the native-run command, which in this case is being called by the capacitor command, so I think including the requirement here in the docs is still a good callout.

File: docs/main/android/troubleshooting.md (hunk 2 above)
Comment (ceceliacreates) [reply docs/main/android/troubleshooting.md]: Updated!

**Review by jcesarmobile — APPROVED** [review]

**Pull request merged**